### PR TITLE
Fixed testing to have comprehensive testing for the library

### PR DESCRIPTION
**Pull request by DGonzalezVillal**
Several bugs caught when trying to run cargo test with different feature combinations. Fixed these bugs and made some slight tweaks to be able to run comprehensive testing on the library.

- When running cargo test with dangerous_hw_tests enabled both has_sev and has_sev_guest tests were run leading to a failure. Fixed by removing an unecessary or
- Renamed has_sev to host and renamed has_sev_guest to guest
- ID-BLOCK had to be put behind openssl only
- Changed snp_config test to hit only the IOCTL and not modify the config. This was done to avoid messing up system configs during testing.
- Renamed the launch test to sev_launch test. This allows us to test that launch test individually.
- Added code to create a legacy SEV cert-chain locally. This allows us to test the SEV launch test without the need of an external tool.
- Cached Turin certificates to the SEV builtin certs and the SNP builtin certs.

### CODE REVIEW

**Review by larrydewey — APPROVED** [review]

**Comment by tylerfanelli** [comment]
The Turin certificate additions that were originally in #242 are also included in this commit. Can you rebase on main and remove the cert additions?

**Comment by DGonzalezVillal** [comment]
Bumped mac-os version in linter

**Comment by DGonzalezVillal** [comment]
@tylerfanelli

File: src/certs/sev/builtin/turin/ark.cert
Comment (tylerfanelli) [inline src/certs/sev/builtin/turin/ark.cert:1]: Were these not added in #242 ? Edit: I see the SNP builtin certs were added. Are they different?

File: src/util/cached_chain.rs [hunk 1]
```diff
@@ -11,21 +11,31 @@
 //! An entire certificate chain can be created using the `sevctl`
 //! utility.
 
-#[cfg(feature = "sev")]
-use crate::certs::sev::Chain;
+#![cfg(all(feature = "sev", feature = "dangerous_hw_tests"))]
```
Comment (tylerfanelli) [inline src/util/cached_chain.rs:14]: What is this for?

**Review by tylerfanelli — COMMENTED** [review]
Two quick comments, and this can be merged.

**Review by tylerfanelli — APPROVED** [review]

**Pull request merged**